OpenRA.Mods.AS/Traits/TeleportNetwork.cs

```diff
@@ -0,0 +1,225 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2015- OpenRA.Mods.AS Developers (see AUTHORS)
+ * This file is a part of a third-party plugin for OpenRA, which is
+ * free software. It is made available to you under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation. For more information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using OpenRA.Mods.AS.Activities;
+using OpenRA.Mods.Common;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Mods.Common.Orders;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.AS.Traits
+{
+	[Desc("This player actor has this many Nydus canals. Must be attached to the player by attaching in player.yaml, otherwise Nydus logic won't work.")]
+	public class TeleportNetworkManagerInfo : TraitInfo<TeleportNetworkManager> { }
+	public class TeleportNetworkManager
+	{
+		public int Count = 0;
+		public Actor PrimaryActor = null;
+
+		[Desc("Type of TeleportNetwork that pairs up, in order for it to work.")]
+		public HashSet<string> Types = new HashSet<string>() { "default" };
+	}
+
+	[Desc("This actor can teleport actors like Nydus canels in SC1. Assuming static object. The actor must have PrimaryBuilding trait so that exit can be specified!")]
+	public class TeleportNetworkInfo : ITraitInfo, IRulesetLoaded
+	{
+		[Desc("Type of TeleportNetwork that pairs up, in order for it to work.")]
+		public HashSet<string> Types = new HashSet<string>() { };
+
+		[Desc("Stances requirement that targeted TeleportNetwork has to meet in order to teleport units.")]
+		public Stance ValidStances = Stance.Ally;
+
+		public object Create(ActorInitializer init) { return new TeleportNetwork(init, this); }
+
+		public void RulesetLoaded(Ruleset rules, ActorInfo info)
+		{ }
+	}
+
+	// The nydus canal does nothing.
+	// The actor teleports itself, upon entering: works the same as EngineerRepairalbe trait.
+	public class TeleportNetwork : INotifyAddedToWorld, INotifyActorDisposing, INotifyOwnerChanged
```

Comment:
INotifyActorDisposing?!

Should be INotifyRemovedFromWorld instead for consistency.